src/wiki/development/build-instructions-linux.md

```diff
@@ -0,0 +1,230 @@
+
+### Build dependencies
+
+- A C++ compiler capable of building C++17 code.
+- Qt Development tools 5.12 or newer (`qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools libqt5core5a libqt5network5 libqt5gui5` on Debian-based systems)
+- Alternatively you can also use Qt 6.0 or newer (`qt6-base-dev qtchooser qt6-base-dev-tools libqt6core6 libqt6core5compat6-dev libqt6network6` on Debian (testing/unstable) based systems), if you prefer it.
+- cmake 3.15 or newer (`cmake` on Debian-based system)
+- ninja (`ninja-build` on Debian-based systems)
+- extra-cmake-modules (`extra-cmake-modules` on Debian-based system)
+- zlib (`zlib1g-dev` on Debian-based system)
+- Java JDK (`openjdk-17-jdk` on Debian-based system)
+- GL headers (`libgl1-mesa-dev` on Debian-based system)
+- scdoc if you want to generate manpages (`scdoc` on Debian-based system)
+
+You can use IDEs, like KDevelop, QtCreator or CLion to open the CMake project, if you want to work on the code.
+
+### Building a portable binary
+
+```bash
+cmake -S . -B build -G Ninja \
+   -DCMAKE_INSTALL_PREFIX=install
+#  -DLauncher_QT_VERSION_MAJOR="6" # if you want to use Qt 6
+
+cmake --build build
+cmake --install build
+cmake --install build --component portable
+```
+
+### Building & installing to the system
+
+This is the preferred method of installation, and is suitable for packages.
+
+```bash
+cmake -S . -B build -G Ninja \
+   -DCMAKE_BUILD_TYPE=Release \
+   -DCMAKE_INSTALL_PREFIX="/usr" \ # Use "/usr" when building Linux packages. If building not for package, use "/usr/local"
+   -DENABLE_LTO=ON # if you want to enable LTO/IPO
+#  -DLauncher_QT_VERSION_MAJOR="6" # if you want to use Qt 6
+
+cmake --build build
+cmake --install build # Optionally specify DESTDIR for packages (i.e. DESTDIR=${pkgdir} cmake --install ...)
+```
+
+### Building a .deb
+
+Requirements: [makedeb](https://docs.makedeb.org/) installed on your system.
+
+```bash
+git clone https://mpr.makedeb.org/prismlauncher.git
+cd prismlauncher
+makedeb -s
+```
+
+The .deb will be located in the directory the repo was cloned in.
+
+### Building an .rpm
+
+Build dependencies are automatically installed using `DNF`, however, you will also need the `rpmdevtools` package (on Fedora),
+in order to fetch sources and set up your tree.
+You don't need to clone the repo for this; the spec file handles that.
+
+```bash
+cd ~
+# setup your ~/rpmbuild directory, required for rpmbuild to work.
+rpmdev-setuptree
+# get the rpm spec file from the prismlauncher-rpm repo
+git clone https://pagure.io/prismlauncher-rpm.git
+cd prismlauncher-rpm
+# install build dependencies
+sudo dnf builddep prismlauncher.spec
+sudo dnf builddep -D "_without_qt6 1" prismlauncher.spec # if you want to use Qt 5 instead of Qt 6
+# download build sources
+spectool -g -R prismlauncher.spec
+# move patch to rpmbuild sources directory
+cp change-jars-path.patch ~/rpmbuild/SOURCES 
```

Comment:
this patch was upstreamed, only one called `0001-find-cmark-with-pkgconfig.patch` is used now. it might be best to make this `cp *.patch ~/rpmbuild/SOURCES` and make the comment something like "copy any patches to rpmbuild sources directory"